fda/grid.py

```diff
@@ -44,6 +45,242 @@ def _list_of_arrays(original_array):
     return list(new_array)
 
 
+class _GridInterpolator:
+
+    def __init__(self, sample_points, data_matrix, ndim_domain, ndim_image, k=1,
+                 s=0., monotone=False, keepdims=False):
+
+        self._ndim_image = ndim_image
+        self._ndim_domain = ndim_domain
+        self._nsamples = data_matrix.shape[0]
+        self._keepdims = keepdims
+
+        if ndim_domain == 1:
+            self._splines = self._construct_spline_1_m(sample_points,
+                                                       data_matrix,
+                                                       k, s, monotone)
+        elif monotone:
+            raise ValueError("Monotone interpolation is only supporte with "
+                             "domain dimension equal to 1.")
+
+        elif ndim_domain == 2:
+            self._splines = self._construct_spline_2_m(sample_points,
+                                                       data_matrix, k, s)
+
+        elif s!=0:
+            raise ValueError("Smoothing interpolation is only supporte with "
+                             "domain dimension up to 2, s should be 0.")
+
+        else:
+            self._splines = self._construct_spline_n_m(sample_points,
+                                                       data_matrix, k)
+
+    def _construct_spline_1_m(self, sample_points, data_matrix, k, s, monotone):
+
+        if k > 5 or k < 1:
+            raise ValueError(f"Invalid degree of interpolation ({k}). Must be "
+                             f"an integer greater than 0 and lower or "
+                             f"equal than 5.")
+
+        if monotone and s != 0:
+            raise ValueError("Smoothing interpolation is not suported with "
+                             "monotone interpolation")
+
+        if monotone and (k == 2 or k == 4):
+            raise ValueError(f"monotone interpolation of degree {k}"
+                             f"not supported.")
+
+        # Monotone interpolation of degree 1 is performed with linear spline
+        if monotone and k == 1: monotone = False
+
+        # Evaluator of splines called in evaluate
+        self._spline_evaluator = lambda spl, t, der: spl(t, der)
```

Comment:
Lambdas should not be used for named functions, use a def instead.

Reply:
Changed in 62c5f51.